superset/sso_auth.py

```diff
@@ -26,23 +27,49 @@ def get_token_contents(token):
 def _find_user(username, sm):
     """extracted from flask_appbuilder.security.manager.BaseSecurityManager.find_user"""
     user = sm.find_user(username)
+   
+    auth_admin_user_list = sm.auth_admin_user_list
+    auth_role_admin = sm.auth_role_admin
+    auth_user_registration_role = sm.auth_user_registration_role
+    role = sm.find_role(auth_user_registration_role)
+    if auth_admin_user_list and username in auth_admin_user_list:
+        role = sm.find_role(auth_role_admin)
+
+
     if not user:
         user = sm.add_user(
                 username= username,
                 first_name= username,
                 last_name=username,
                 email=username + '@email.notfound',
-                role=sm.find_role("Admin")
+                role=role
             )
+    else:
+        is_role_exists = sm.is_role_exists(role.name,user.roles)
+        if not is_role_exists:
+            user.roles.append(role)
+            sm.update_user(user)
+
     return user
 
 def parse_hadoop_jwt():
-    if(config.IS_KNOX_SSO_ENABLED):
+    logging.debug("----------------Before Request Headers %s", request.headers)
```

Comment:
Again I know you must be taking care of cleanup, but just keeping it as record so it won't get missed.